### ISSUE TITLE
<Image /> with GIF breaks when resizeMode is set to "repeat"

### ISSUE THREAD

**Issue by l1m2p3** [facebook/react-native#42132]
### Description

`<Image />` cannot render GIF correctly when (Android) or after (iOS) setting `resizeMode` to `"repeat"`.

The issue on iOS and Android are different:
1. on iOS,
1.1. the first issue is GIF cannot animate when `resizeMode` is set to `"repeat"`
1.2. the second issue is `<Image />` cannot get out of `"repeat"` mode after setting `resizeMode` to it. 
2. on Android, the issue is `<Image />` cannot render `"repeat"` mode correctly -- there GIF will not "repeat", but shrink to top left corner instead

JPG and PNG files don't have any issue.

### Steps to reproduce

Setup: create a vanilla React Native app, add a `Image` component in it. Or try the linked Expo Snack below

1. try JPG and PNG with various resizeMode value, including "repeat", and observe JPG and PNG are always rendered correctly
2. try the same with GIF, and observe:
2.1. on iOS, GIF cannot animate when `resizeMode` is set to "repeat"
2.2. on iOS, GIF cannot recover from "repeat" after setting `resizeMode` to "repeat" and then other value (e.g. "cover")
2.2. on Android, GIF cannot "repeat", but shrinks to the top left corner instead

### React Native Version

0.72.6

### Affected Platforms

Runtime - Android, Runtime - iOS

### Output of `npx react-native info`

```text
N/A
```

### Stacktrace or Logs

```text
N/A
```

### Reproducer

https://snack.expo.dev/@l1m2p3/image-gif-resizemode-bug

### Screenshots and Videos

**iOS:** 

https://github.com/facebook/react-native/assets/6405712/9d2795bd-79f1-444e-8815-228db3446855

**Android:** 

https://github.com/facebook/react-native/assets/6405712/ca4e803e-56bc-4f58-9fb6-77bfa03279eb

**Comment by cortinico** [comment]
Can we try if this still happens on 0.73?

**Comment by l1m2p3** [comment]
> Can we try if this still happens on 0.73?

Yeah. I updated RN and Expo to use RN v0.73.1, but both iOS and Android can still repro the issue.

**Comment by l1m2p3** [comment]
> on android, `.gif` cannot play in `0.73.6` no matter how `resizeMode` is.

That could be a different issue. See if you missed this in your Android project's .gradle file?
https://reactnative.dev/docs/0.71/image#gif-and-webp-support-on-android

**Comment by neeraj500** [comment]
> on android, `.gif` cannot play in `0.73.6` no matter how `resizeMode` is.

yes, gif is not working on 0.73.6
